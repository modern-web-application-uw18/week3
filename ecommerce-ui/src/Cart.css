.Cart {
    font-size: 1em;
    color: Grey;
    align: left;
    margin: 20px;

}

.Cart__button {
    border: none;
    font-size: 2em;
    padding: 10px 30px;
    margin: 10px;
    cursor: pointer;
}

.Cart__button--addroom {
    align: right;
    color: red;
}

.Cart__button--removeroom {
    background-color: red;
}

.Cart__button--clear {
    

}