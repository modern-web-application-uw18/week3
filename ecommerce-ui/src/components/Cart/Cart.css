.Cart {
    background-color: rgb(95, 95, 95);
    height: 50px;
    font-family: 'Roboto', sans-serif;
    font-weight: 300;
    font-style: italic;
    font-size: 40px;
    color: #ffffff;
    margin-bottom: 0;
    text-align: right;

}


@media (max-width: 650px) {
    .Cart {
        font-size: 30px;
    }
}