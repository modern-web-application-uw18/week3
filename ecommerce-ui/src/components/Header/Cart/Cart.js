import React from 'react';
import Modal from 'react-bootstrap/lib/Modal';
import Button from 'react-bootstrap/lib/Button';
import MdHighlightRemove from 'react-icons/lib/md/highlight-remove';
import PropTypes from 'prop-types';



const cart = (props) => (
    

    <Modal show={props.cartShow} onHide={props.handleCartClose}>
        <Modal.Header closeButton>
            <Modal.Title>Shopping Cart</Modal.Title>
        </Modal.Header>
        <Modal.Body>


   

            <table className="table table-hover">
                <thead>
                    <tr>
                        <th></th>
                        <th scope="col">Title</th>
                        <th scope="col">City</th>
                        <th scope="col">Country</th>
                        <th scope="col">Cost</th>
                    </tr>
                </thead>
                <tbody>


                    {props.cartContent.map((item, idx) => {
                        return (
                        <tr key={idx}>
                            <td><MdHighlightRemove style={{color:'red'}} onClick={()=>props.removeListingFromTheCartHandler(item)}/></td>
                            <td>{item.title}</td>
                            <td>{item.location.city}</td>
                            <td>{item.location.country}</td>
                            <td>$ {item.payment.cost}</td>
                        
                        </tr>)
                    })}

                    <tr>
                        <td/>
                        <td/>
                        <td/>
                        <td><b>Total</b></td>
                        <td>$ {props.cartContent.reduce((sum,value) => sum+value.payment.cost, 0)}</td>
                    </tr>


                </tbody>
            </table>


        </Modal.Body>
        <Modal.Footer>
            <Button onClick={props.handleCartClose}>Close</Button>
        </Modal.Footer>
    </Modal>
)


cart.propTypes = {
    cartShow: PropTypes.bool.isRequired,
    handleCartClose: PropTypes.func.isRequired,
    cartContent: PropTypes.array.isRequired,
    removeListingFromTheCartHandler: PropTypes.func.isRequired
}


export default cart;
