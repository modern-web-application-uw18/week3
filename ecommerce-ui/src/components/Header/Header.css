.Header {
    background-color: #000000;
    width: 100%;
    height: 65px;
    font-family: 'Roboto', sans-serif;
    font-weight: 300;
    font-style: italic;
    font-size: 40px;
    color: #ffffff;
    margin-bottom: 0;
    padding-right: 10px;
    padding-left: 10px;
    display: grid;
    justify-content: space-between;
    grid-template-columns: 50% 50%;
    grid-auto-rows: 50%;
    grid-gap: 5px;


}


@media (max-width: 650px) {
    .Header {
        font-size: 30px;
        padding-top: 10px;
    }
}