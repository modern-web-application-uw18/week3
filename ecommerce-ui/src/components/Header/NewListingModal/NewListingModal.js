import React, { Component } from 'react';
import Modal from 'react-bootstrap/lib/Modal';
import Button from 'react-bootstrap/lib/Button';
import Form from 'react-bootstrap/lib/Form';
import FormGroup from 'react-bootstrap/lib/FormGroup';
import Col from 'react-bootstrap/lib/Col';
import FormControl from 'react-bootstrap/lib/FormControl';
import PropTypes from 'prop-types';


class NewListingModal extends Component {


    state = {
        title: '',
        houseType: '',
        image: '',
        city: '',
        country: '',
        cost: '',
        name: ''
    }


    handleNameChange = (e) => {
        let name = e.target.value;
        this.setState((prevState, props) => {
            return { name: name };
        });
    }

    handleCityChange = (e) => {
        let city = e.target.value;
        this.setState((prevState, props) => {
            return { city: city };
        });
    }

    handleCountryChange = (e) => {
        let country = e.target.value;
        this.setState((prevState, props) => {
            return { country: country };
        });
    }

    handleTitleChange = (e) => {
        let title = e.target.value;
        this.setState((prevState, props) => {
            return { title: title };
        });
    }

    handleHouseTypeChange = (e) => {
        let houseType = e.target.value;
        this.setState((prevState, props) => {
            return { houseType: houseType };
        });
    }

    handleImageURLChange = (e) => {
        let image = e.target.value;
        this.setState((prevState, props) => {
            return { image: image };
        });
    }

    handleCostChange = (e) => {
        let cost = e.target.value;
        this.setState((prevState, props) => {
            return { cost: cost };
        });
    }


    submitForm = () => {

        let proceed = true;

        this.props.mockDataArr.filter(
            (obj, idx) => {
                if (obj.title === this.state.title) {
                    proceed = false;
                    alert('Title already exists. Please choose another one!');
                    this.setState((prevState, props) => {
                        return { title: '' };
                    });
                }
                return null;
            }
        );

        if (this.getValidationState() === 'success' && proceed) {

            this.props.dataHandler(this.state);
            this.props.handleFormClose();
            this.setState((prev, props) => {
                return {
                    title: '',
                    houseType: '',
                    image: '',
                    city: '',
                    country: '',
                    cost: '',
                    name: ''
                }
            });
        }

    }

    getValidationState = () => {


        const length = this.state.title.length;
        if (length > 0) return 'success';
        else if (length > 0) return 'warning';
        else if (length === 0) return 'error';
        return null;
    }

    render() {
        return (
            <Modal show={this.props.formShow} onHide={this.props.handleFormClose}>
                <Modal.Header closeButton>
                    <Modal.Title>Add a New Listing</Modal.Title>
                </Modal.Header>
                <Modal.Body>

                    <Form horizontal>

                        <FormGroup controlId="formTitle" validationState={this.getValidationState()}>
                            <Col sm={3}>Title</Col>
                            <Col sm={9}>
                                <FormControl type="text" value={this.state.title} onChange={this.handleTitleChange} placeholder="Title" />
                            </Col>
                        </FormGroup>

                        <FormGroup controlId="formName">
                            <Col sm={3}>Name</Col>
                            <Col sm={9}>
                                <FormControl type="text" value={this.state.name} onChange={this.handleNameChange} placeholder="Name" />
                            </Col>
                        </FormGroup>

                        <FormGroup controlId="formCity">
                            <Col sm={3}>City</Col>
                            <Col sm={9}>
                                <FormControl type="text" value={this.state.city} onChange={this.handleCityChange} placeholder="City" />
                            </Col>
                        </FormGroup>


                        <FormGroup controlId="formCountry">
                            <Col sm={3}>Country</Col>
                            <Col sm={9}>
                                <FormControl type="text" value={this.state.country} onChange={this.handleCountryChange} placeholder="Country" />
                            </Col>
                        </FormGroup>

                        <FormGroup controlId="formHouseType">
                            <Col sm={3}>House Type</Col>
                            <Col sm={9}>
                                <FormControl componentClass="select" value={this.state.houseType} onChange={this.handleHouseTypeChange} placeholder="Select">
                                    <option value="">...</option>
                                    <option value="Apartment">Apartment</option>
                                    <option value="Entire House">Entire House</option>
                                </FormControl>
                            </Col>
                        </FormGroup>

                        <FormGroup controlId="formImage">
                            <Col sm={3}>Image URL</Col>
                            <Col sm={9}>
                                <FormControl type="text" value={this.state.imageURL} onChange={this.handleImageURLChange} placeholder="Image URL" />
                            </Col>
                        </FormGroup>

                        <FormGroup controlId="formCost">
                            <Col sm={3}>Cost</Col>
                            <Col sm={9}>
                                <FormControl type="number" value={this.state.cost} onChange={this.handleCostChange} placeholder="Cost" />
                            </Col>
                        </FormGroup>

                        <FormGroup>
                            <Col smOffset={3} sm={9}>
                                <Button className="submitBtn" onClick={this.submitForm}>Submit New Listing</Button>
                            </Col>
                        </FormGroup>

                    </Form>


                </Modal.Body>
                <Modal.Footer>
                    <Button onClick={this.props.handleFormClose}>Close</Button>

                </Modal.Footer>
            </Modal>
        )
    }


}


NewListingModal.propTypes = {
    formShow: PropTypes.bool.isRequired,
    handleFormClose: PropTypes.func.isRequired,
    dataHandler: PropTypes.func.isRequired,
    mockDataArr: PropTypes.array.isRequired
}

export default NewListingModal;