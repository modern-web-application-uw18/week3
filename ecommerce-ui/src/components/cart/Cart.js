import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './cart.css';

class Cart extends Component {
  render() {

    const { length } = this.props.inCart;

    const cartTotal = () => {
      let total = 0;
      for (let rental of this.props.inCart) {
        total += rental.payment.cost
      }

      return total;
    }

    return (
      <aside className="cart-container"  id="cart-title">
        <h2 className="cart-container__title">Requested Rentals</h2>

        {length === 0 &&
          <div className="cart-item__total">You have no selections</div>
        }

        <div className="cart-item__container">
          {this.props.inCart.map((rental, index) =>
            <div key={index} className="cart-item">
              <h3 className="cart-item__title">{rental.title}</h3>
              <p className="cart-item__cost">${rental.payment.cost}</p>
              <button className="btn cart-item__btn" onClick={this.props.deleteRental(index)}>Remove</button>
            </div>
          )}

          {length !== 0 &&
            <div>
              <h2 className="cart-container__title">Request Total</h2>
              <div className="cart-item__total">
                {length} rental{length === 1 ? '' : 's'} requested | <strong>Total ${cartTotal()}</strong>
              </div>
            </div>
          }

        </div>
      </aside>
    );
  }
}

Cart.propTypes = {
  deleteRental: PropTypes.func.isRequired
}

export default Cart;
