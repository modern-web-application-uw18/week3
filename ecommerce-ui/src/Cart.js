import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './Cart.css';

class Cart extends Component {

    constructor(props) {
        super(props);
        this.removeRoomAction = this.props.removeRoomAction.bind(this);
      }
    
    render() {
        let removeRoomAction = this.removeRoomAction;
        return (
            // <div className="Cart">
            //     <h2>{this.state.value}</h2>
            //     <button className="Cart____button--removeroom" onClick={this.decrement}>Remove Room</button>
            //     <button className="Cart__button--addroom" onClick={this.increment}>Add Room</button>
            // </div>
            <div class="Cart">
            <h1>Saved Rooms</h1>
            <ul>
            {this.props.rooms.map(function(room, index){
                return (<li key={ index }>{room.title}&nbsp;&nbsp;${room.payment.cost}&nbsp;&nbsp;<button type="button" onClick={() => removeRoomAction(room)}>Delete Room</button></li>);
              })}
            </ul>
            </div>
        );
    }
}

Cart.propTypes = {
    removeRoomAction: PropTypes.func,
    rooms: PropTypes.array
};

Cart.defaultProps = {
    rooms: []
};

export default Cart;