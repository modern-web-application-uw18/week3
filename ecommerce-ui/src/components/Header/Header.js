import React, { Component } from 'react';
import classes from './Header.css';
import Cart from './Cart/Cart';
import NewListingModal from './NewListingModal/NewListingModal';

import MdShoppingCart from 'react-icons/lib/md/shopping-cart';
import MdAddBox from 'react-icons/lib/md/add-box';

class Header extends Component {
  state = {
    cartShow: false,
    formShow: false
  }

  handleCartClose = () => {
    this.setState({ cartShow: false });
  }

  handleCartShow = () => {
    this.setState({ cartShow: true });
  }


  handleFormClose  = () => {
    this.setState({ formShow: false });
  }

  handleFormShow  = () => {
    this.setState({ formShow: true });
  }

  render() {

    return (
      <div className={classes.Header}>

        <div>
          <button type="button" className="btn btn-danger" style={{ width: '150px', height: '40px', border: '0px' }} onClick={this.handleFormShow}>
            Add Listing &nbsp;&nbsp; <MdAddBox size={30} />
          </button>
        </div>

        <div style={{ textAlign: 'right' }}>
          <button type="button" className="btn btn-default" style={{ width: '150px', height: '40px', border: '0px' }} onClick={this.handleCartShow}>
            Checkout &nbsp;&nbsp; <MdShoppingCart size={30} />
          </button>
        </div>


        <Cart cartShow={this.state.cartShow}  handleCartClose={this.handleCartClose} cartContent={this.props.cartContent} removeListingFromTheCartHandler={this.props.removeListingFromTheCartHandler} />
        <NewListingModal formShow={this.state.formShow} handleFormClose={this.handleFormClose} dataHandler={this.props.dataHandler} mockDataArr={this.props.mockDataArr}/>


      </div>
    )
  }
}


export default Header;
