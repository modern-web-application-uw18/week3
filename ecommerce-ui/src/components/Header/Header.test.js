import React from 'react';
import { shallow, mount} from 'enzyme';
import Header from './Header';
import classes from './Header.css';
import Cart from './Cart/Cart';
import NewListingModal from './NewListingModal/NewListingModal';

import MdShoppingCart from 'react-icons/lib/md/shopping-cart';
import MdAddBox from 'react-icons/lib/md/add-box';

describe('<Header />', () => {

    let mockListing = {
        "title": "Centrally-located Manhattan studio",
        "houseType": "Apartment",
        "image": "https://images.unsplash.com/photo-1490806230066-f7e6f7bf5052?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=2ca889535ea01f912f94ac4ddf0034e0&auto=format&fit=crop&w=500&q=80",
        "location": {
            "city": "New York",
            "country": "USA"
        },
        "payment": {
            "cost": 105,
            "description": "Free cancellation"
        },
        "host": {
            "name": "Robert",
            "isSuperhost": true
        },
        "rating": {
            "stars": 3,
            "reviews": 133
        }
    }

    it('Snapshot test', () => {
        const wrapper = mount(<Header cartContent={[mockListing]}/>);

        //  wrapper.instance().handleCartClose();
        //  wrapper.instance().handleCartShow();
        //  wrapper.instance().handleFormClose();
        //  wrapper.instance().handleFormShow();


        expect(wrapper).toMatchSnapshot();
    })
})