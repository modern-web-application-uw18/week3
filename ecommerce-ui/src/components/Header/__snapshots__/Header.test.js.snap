// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Snapshot test 1`] = `
<Header
  cartContent={
    Array [
      Object {
        "host": Object {
          "isSuperhost": true,
          "name": "Robert",
        },
        "houseType": "Apartment",
        "image": "https://images.unsplash.com/photo-1490806230066-f7e6f7bf5052?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=2ca889535ea01f912f94ac4ddf0034e0&auto=format&fit=crop&w=500&q=80",
        "location": Object {
          "city": "New York",
          "country": "USA",
        },
        "payment": Object {
          "cost": 105,
          "description": "Free cancellation",
        },
        "rating": Object {
          "reviews": 133,
          "stars": 3,
        },
        "title": "Centrally-located Manhattan studio",
      },
    ]
  }
>
  <div>
    <div>
      <button
        className="btn btn-danger"
        onClick={[Function]}
        style={
          Object {
            "border": "0px",
            "height": "40px",
            "width": "150px",
          }
        }
        type="button"
      >
        Add Listing    
        <MdAddBox
          size={30}
        >
          <IconBase
            size={30}
            viewBox="0 0 40 40"
          >
            <svg
              fill="currentColor"
              height={30}
              preserveAspectRatio="xMidYMid meet"
              style={
                Object {
                  "color": undefined,
                  "verticalAlign": "middle",
                }
              }
              viewBox="0 0 40 40"
              width={30}
            >
              <g>
                <path
                  d="m28.4 21.6v-3.2h-6.8v-6.8h-3.2v6.8h-6.8v3.2h6.8v6.8h3.2v-6.8h6.8z m3.2-16.6c1.8 0 3.4 1.6 3.4 3.4v23.2c0 1.8-1.6 3.4-3.4 3.4h-23.2c-1.9 0-3.4-1.6-3.4-3.4v-23.2c0-1.8 1.5-3.4 3.4-3.4h23.2z"
                />
              </g>
            </svg>
          </IconBase>
        </MdAddBox>
      </button>
    </div>
    <div
      style={
        Object {
          "textAlign": "right",
        }
      }
    >
      <button
        className="btn btn-default"
        onClick={[Function]}
        style={
          Object {
            "border": "0px",
            "height": "40px",
            "width": "150px",
          }
        }
        type="button"
      >
        Checkout    
        <MdShoppingCart
          size={30}
        >
          <IconBase
            size={30}
            viewBox="0 0 40 40"
          >
            <svg
              fill="currentColor"
              height={30}
              preserveAspectRatio="xMidYMid meet"
              style={
                Object {
                  "color": undefined,
                  "verticalAlign": "middle",
                }
              }
              viewBox="0 0 40 40"
              width={30}
            >
              <g>
                <path
                  d="m28.4 30c1.8 0 3.2 1.6 3.2 3.4s-1.4 3.2-3.2 3.2-3.4-1.4-3.4-3.2 1.6-3.4 3.4-3.4z m-26.8-26.6h5.5l1.6 3.2h24.7c0.9 0 1.6 0.8 1.6 1.8 0 0.3-0.1 0.5-0.2 0.7l-6 10.8c-0.5 1-1.6 1.7-2.9 1.7h-12.4l-1.5 2.8 0 0.2c0 0.2 0.1 0.4 0.3 0.4h19.3v3.4h-20c-1.8 0-3.2-1.6-3.2-3.4 0-0.5 0.1-1.1 0.3-1.6l2.3-4.1-6-12.7h-3.4v-3.2z m10 26.6c1.8 0 3.4 1.6 3.4 3.4s-1.6 3.2-3.4 3.2-3.2-1.4-3.2-3.2 1.4-3.4 3.2-3.4z"
                />
              </g>
            </svg>
          </IconBase>
        </MdShoppingCart>
      </button>
    </div>
    <cart
      cartContent={
        Array [
          Object {
            "host": Object {
              "isSuperhost": true,
              "name": "Robert",
            },
            "houseType": "Apartment",
            "image": "https://images.unsplash.com/photo-1490806230066-f7e6f7bf5052?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=2ca889535ea01f912f94ac4ddf0034e0&auto=format&fit=crop&w=500&q=80",
            "location": Object {
              "city": "New York",
              "country": "USA",
            },
            "payment": Object {
              "cost": 105,
              "description": "Free cancellation",
            },
            "rating": Object {
              "reviews": 133,
              "stars": 3,
            },
            "title": "Centrally-located Manhattan studio",
          },
        ]
      }
      cartShow={false}
      handleCartClose={[Function]}
    >
      <Modal
        animation={true}
        autoFocus={true}
        backdrop={true}
        bsClass="modal"
        dialogComponentClass={[Function]}
        enforceFocus={true}
        keyboard={true}
        manager={
          ModalManager {
            "add": [Function],
            "containers": Array [],
            "data": Array [],
            "handleContainerOverflow": true,
            "hideSiblingNodes": true,
            "isTopModal": [Function],
            "modals": Array [],
            "remove": [Function],
          }
        }
        onHide={[Function]}
        renderBackdrop={[Function]}
        restoreFocus={true}
        show={false}
      >
        <Modal
          autoFocus={true}
          backdrop={true}
          backdropClassName="modal-backdrop"
          backdropTransition={[Function]}
          containerClassName="modal-open"
          enforceFocus={true}
          keyboard={true}
          manager={
            ModalManager {
              "add": [Function],
              "containers": Array [],
              "data": Array [],
              "handleContainerOverflow": true,
              "hideSiblingNodes": true,
              "isTopModal": [Function],
              "modals": Array [],
              "remove": [Function],
            }
          }
          onEntering={[Function]}
          onExited={[Function]}
          onHide={[Function]}
          renderBackdrop={[Function]}
          restoreFocus={true}
          show={false}
          transition={[Function]}
        />
      </Modal>
    </cart>
    <NewListingModal
      formShow={false}
      handleFormClose={[Function]}
    >
      <Modal
        animation={true}
        autoFocus={true}
        backdrop={true}
        bsClass="modal"
        dialogComponentClass={[Function]}
        enforceFocus={true}
        keyboard={true}
        manager={
          ModalManager {
            "add": [Function],
            "containers": Array [],
            "data": Array [],
            "handleContainerOverflow": true,
            "hideSiblingNodes": true,
            "isTopModal": [Function],
            "modals": Array [],
            "remove": [Function],
          }
        }
        onHide={[Function]}
        renderBackdrop={[Function]}
        restoreFocus={true}
        show={false}
      >
        <Modal
          autoFocus={true}
          backdrop={true}
          backdropClassName="modal-backdrop"
          backdropTransition={[Function]}
          containerClassName="modal-open"
          enforceFocus={true}
          keyboard={true}
          manager={
            ModalManager {
              "add": [Function],
              "containers": Array [],
              "data": Array [],
              "handleContainerOverflow": true,
              "hideSiblingNodes": true,
              "isTopModal": [Function],
              "modals": Array [],
              "remove": [Function],
            }
          }
          onEntering={[Function]}
          onExited={[Function]}
          onHide={[Function]}
          renderBackdrop={[Function]}
          restoreFocus={true}
          show={false}
          transition={[Function]}
        />
      </Modal>
    </NewListingModal>
  </div>
</Header>
`;
